### ISSUE TITLE
[Bug/Feature Request] Mismtached Songs

### ISSUE THREAD

**Issue by TheEmpty** [Yooooomi/your_spotify#338]
**Describe the bug**

Listening to Bittersweet from the Bittersweet album by Foxblood, https://open.spotify.com/track/6MRX2sQeyskNaHkCeIhmUl?si=ca1995dfd41e48ef
It shows up as Bittersweet 2.0 from the Reprise album by Foxblood, https://open.spotify.com/track/7ws2xqXw50IS1q1JTIhVuK?si=8c59c9b60c164acb

**Expected behavior**

Best: Matches to the non-2.0 version.
New Feature/General request: "Fix this match" similar to how Plex does when it's matching incorrectly tags stuff. Unsure if DB today stores enough information to fix individual plays, but from "Recently played on" seems like it could unless that data is purged at some point.

**Additional context**

Version 1.7.3 on Docker compose in LXC on Proxmox. Imported the shorter Spotify dataset, still waiting on the big one.

**Screenshots**

In your_spotify shows as only 1 play,
<img width="386" alt="Screenshot 2024-01-22 at 10 42 14 AM" src="https://github.com/Yooooomi/your_spotify/assets/371940/15b4f99c-2adf-49cb-bc95-06e9e3ae9993">

But this was my top song of 2023,

<img width="727" alt="Screenshot 2024-01-22 at 10 42 29 AM" src="https://github.com/Yooooomi/your_spotify/assets/371940/1bc4b529-d3c1-4939-bb09-ecd1bf25ffb7">

**Shared link**

https://your-spotify.marcopolo.best/song/7ws2xqXw50IS1q1JTIhVuK?gname=All&token=6029bccf-06b1-4720-b3d2-f5e8cc08eb88

also data files so others can replicate if needed,

[StreamingHistory_music_1.json](https://github.com/Yooooomi/your_spotify/files/14012331/StreamingHistory_music_1.json)
[StreamingHistory_music_0.json](https://github.com/Yooooomi/your_spotify/files/14012332/StreamingHistory_music_0.json)

**Comment by TheEmpty** [comment]
Okay hmm. Saw in another issue that the extended version has IDs so it’s not trying to string match like this API is. Today are you just taking the first result versus looking at like the first batch of results and check nearest match? But yeah I might just delete my account and recreate it when the long version comes in then so I can correct things. Hopefully no missing data since the short one was to the day anyway and not from request date.

**Comment by TheEmpty** [comment]
Deleted account on my instance so share link won't work.

Deleted as the extended data came in. It only has data up to the date that I requested it. So I'll have to request it again later to get the past 5 days of data not included. Since these all have ID I should be able to import the next export and have correct statistics. 

I wonder if the recent history import should either be removed or tagged so that data can be removed eaisly.

**Comment by Yooooomi** [comment]
Hello! If you request all the data from the creation of your account I would suggest that you just remove your instance and import from a new account. Also, the import handles duplicates so you should not have issues importing on top.
